access/services/v2/groups.go

```diff
@@ -0,0 +1,146 @@
+	GroupDetails
+}
+
+type GroupDetails struct {
+	Name            string   `json:"name,omitempty"`
+	Description     string   `json:"description,omitempty"`
+	AutoJoin        *bool    `json:"auto_join,omitempty"`
+	AdminPrivileges *bool    `json:"admin_privileges,omitempty"`
+	Realm           string   `json:"realm,omitempty"`
+	RealmAttributes string   `json:"realm_attributes,omitempty"`
+	ExternalId      string   `json:"external_id,omitempty"`
+	Members         []string `json:"members,omitempty"`
+}
+
+type GroupListItem struct {
+	GroupName string `json:"group_name"`
+	Uri       string `json:"uri"`
+}
+type GroupList struct {
+	Cursor string          `json:"cursor,omitempty"`
+	Groups []GroupListItem `json:"groups"`
+}
+
+func NewGroupParams() GroupParams {
+	return GroupParams{}
+}
+
+type GroupService struct {
+	client         *jfroghttpclient.JfrogHttpClient
+	ServiceDetails auth.ServiceDetails
+}
+
+func NewGroupService(client *jfroghttpclient.JfrogHttpClient) *GroupService {
+	return &GroupService{client: client}
+}
+
+func (gs *GroupService) getBaseUrl() string {
+	return fmt.Sprintf("%s%s", gs.ServiceDetails.GetUrl(), groupsApi)
+}
+func (gs *GroupService) GetAll() ([]GroupListItem, error) {
+	httpDetails := gs.ServiceDetails.CreateHttpClientDetails()
+	url := gs.getBaseUrl()
+	resp, body, _, err := gs.client.SendGet(url, true, &httpDetails)
+	if err != nil {
+		return nil, err
+	}
+
+	if err = errorutils.CheckResponseStatusWithBody(resp, body, http.StatusOK); err != nil {
+		return nil, err
+	}
+	var groupList GroupList
+	err = json.Unmarshal(body, &groupList)
+	return groupList.Groups, errorutils.CheckError(err)
+}
+
+func (gs *GroupService) Get(name string) (u *GroupDetails, err error) {
+	httpDetails := gs.ServiceDetails.CreateHttpClientDetails()
+	url := fmt.Sprintf("%s/%s", gs.getBaseUrl(), name)
+	resp, body, _, err := gs.client.SendGet(url, true, &httpDetails)
+	if err != nil {
+		return nil, err
+	}
+
+	if resp.StatusCode == http.StatusNotFound {
+		return nil, nil
+	}
+	if err = errorutils.CheckResponseStatusWithBody(resp, body, http.StatusOK); err != nil {
+		return nil, err
+	}
+	var group GroupDetails
+	err = json.Unmarshal(body, &group)
+	return &group, errorutils.CheckError(err)
+}
+
+func (gs *GroupService) Create(params GroupParams) error {
+	group, err := gs.Get(params.Name)
+	if err != nil {
+		return err
+	}
+	if group != nil {
+		return errorutils.CheckErrorf("group '%s' already exists", group.Name)
+	}
+	content, httpDetails, err := gs.createOrUpdateRequest(params)
+	if err != nil {
+		return err
+	}
+	resp, body, err := gs.client.SendPost(gs.getBaseUrl(), content, &httpDetails)
+	if err != nil {
+		return err
+	}
+	return errorutils.CheckResponseStatusWithBody(resp, body, http.StatusCreated)
+}
+
+func (gs *GroupService) Update(params GroupParams) error {
+	content, httpDetails, err := gs.createOrUpdateRequest(params)
+	if err != nil {
+		return err
+	}
+	url := fmt.Sprintf("%s/%s", gs.getBaseUrl(), params.Name)
```

Comment:
You can use `path.Join` in this case. The same goes for a few other places in this PR.